### DISCUSSION TITLE
why api /sb/keywords and /sb/targets/list can not get total data?

### DISCUSSION THREAD

**Discussion by LionSniper** [amzn/ads-advanced-tools-docs#294] [Q&A]
when i request api /sb/keywords, i found a problem, i can not get total keywords, only when I add the campaignIdFilter filter condition can I request the keywords under this campaign. For example: i have a brand video ad_product, and this product it's campaign and contains keywords, but I cannot retrieve the data for this keyword unless I add the filter condition campaignIdFilter=xxxxx (the state field filtering already includes everything). The same issue applies to the API /sb/targets/list

Please tell me the reason and how to solve it. Thank you  @elizafarley @kdwivedi13

**Comment by elizafarley** [comment]
Hi @LionSniper, this may be a technical bug with this API. I'd suggest submitting a support ticket if you haven't already.

You could also try using the [Exports API](https://advertising.amazon.com/API/docs/en-us/exports) to pull this data. Exports will return all targeting expressions without the need for any filters.
